framework/basehandlers.py

```diff
@@ -413,10 +413,13 @@ def split_emails(self, param_name):
   def parse_link(self, param_name):
     link = flask.request.form.get(param_name) or None
     if link:
-      if not link.startswith('http'):
-        link = str('http://' + link)
-      else:
-        link = str(link)
+      # see https://www.regextester.com/93901 for url regex
+      url_pattern = (r'\b(http:\/\/|https:\/\/)?([\w_-]+(?:(?:\.[\w_-]+)+))'
+                     r'([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])?\b')
+      match_obj = re.search(url_pattern, str(link))
+      link = match_obj.group() or None
+      if not link.startswith(('http://', 'https://')):
```

Comment:
This will cause an error if link is None.

Reply:
I made a slight change, now the script would check if there's actually a ```match_obj``` before moving on to process it.